spec/loophp/collection/CollectionSpec.php

```diff
@@ -1822,28 +1825,51 @@ public function it_can_map(): void
         $input = array_combine(range('A', 'E'), range('A', 'E'));
 
         $this::fromIterable($input)
-            ->map(static function (string $item): string {
-                return $item . $item;
-            })
+            ->map(static fn (string $item): string => $item . $item)
             ->shouldIterateAs(['A' => 'AA', 'B' => 'BB', 'C' => 'CC', 'D' => 'DD', 'E' => 'EE']);
 
-        $callback1 = static function (string $a): string {
-            return $a . $a;
-        };
+        $square = static fn (int $a): int => $a ** 2;
+        $toString = static fn (int $a): string => (string) $a;
+        $appendBar = static fn (string $a): string => $a . 'bar';
 
-        $callback2 = static function (string $a): string {
-            return '[' . $a . ']';
-        };
+        $this::fromIterable(range(1, 3))
+            ->map($square)
+            ->map($toString)
+            ->map($appendBar)
+            ->shouldIterateAs(['1bar', '4bar', '9bar']);
 
-        $this::fromIterable(range('a', 'e'))
-            ->map($callback1, $callback2)
-            ->shouldIterateAs([
-                '[aa]',
-                '[bb]',
-                '[cc]',
-                '[dd]',
-                '[ee]',
-            ]);
+        // Note this will work in PHP 8 but fails during testing due to: https://github.com/phpspec/phpspec/issues/1375
+        if (PHP_VERSION_ID >= 80000) {
+            $this::fromIterable(range(1, 3))
+                ->map($square, $toString)
+                ->shouldThrow(ErrorException::class)->during('all');
+        } else {
+            $this::fromIterable(range(1, 3))
+                ->map($square, $toString)
+                ->shouldIterateAs(['1', '4', '9']);
+        }
+    }
+
+    public function it_can_mapN(): void
+    {
+        $input = array_combine(range('A', 'E'), range('A', 'E'));
+
+        $this::fromIterable($input)
+            ->mapN(static fn (string $item): string => $item . $item)
+            ->shouldIterateAs(['A' => 'AA', 'B' => 'BB', 'C' => 'CC', 'D' => 'DD', 'E' => 'EE']);
+
+        $square = static fn (int $a): int => $a ** 2;
+        $toString = static fn (int $a): string => (string) $a;
+
+        $this::fromIterable(range(1, 3))
+            ->mapN($square, $toString)
+            ->shouldIterateAs(['1', '4', '9']);
+
+        $this::fromIterable(range(1, 3))
+            ->map(static fn (int $item): int => $item ** 2)
```

Comment:
You're using `::map()` and we are in `it_can_mapN`

Reply:
good spot, copy-paste mistake 😅 . Will fix